pkg/api/validation/schema.go

```diff
@@ -135,25 +128,31 @@ func (s *SwaggerSchema) ValidateBytes(data []byte) error {
 	if !ok {
 		return fmt.Errorf("error in unmarshaling data %s", string(data))
 	}
-	groupVersion := fields["apiVersion"]
-	if groupVersion == nil {
+	groupVersionField := fields["apiVersion"]
```

Comment:
Try decoding with `UnstructuredJSONScheme` (https://github.com/kubernetes/kubernetes/blob/master/pkg/runtime/unstructured.go#L29) instead of `json.Unmarshal`.  Anything that isn't `runtime.Unstructured` will never validate, so you can fail there.

Then you'll be able to use the `TypeMeta` from the `Unstructured` instead the lookup and cast approach below.